Coverage

niklasmohrin opened the pull request:
Feel free to push all the fixes about coverage to this branch :)

niklasmohrin committed:
remove unused method

niklasmohrin committed:
isEmpty not => notEmpty

niklasmohrin committed:
add another indirection on ALGWindowOperationsType for accessing `TheWorldMainDockingBar instance` and `Project current world`

niklasmohrin committed:
add test that mocks current project and docking bar buttons so that we can test these operations

niklasmohrin committed:
categorize all uncategorized

niklasmohrin committed:
add docking bar indirection to ALGWindowType as well

niklasmohrin committed:
add test that mocks docking bar for ALGWindowType

Olliwehr committed:
Merge branch 'development' into coverage

Olliwehr commented on packages/AlgernonTests.package/ALGMockedObject.class/properties.json:
```diff
@@ -0,0 +1,15 @@
+	"instvars" : [
+		"mockedMethods",
+		"calls" ],
+	"name" : "ALGMockedObject",
```
Would it be better to also rename your new mocked objects to use "Fake" instead of "Mocked" (as the other mocked things in the testing package follow this naming convention)? But that is just nitpicking tbh :D

niklasmohrin replied:
Hm, I think when coming up with the names, I felt that classes like `FakeKeyboardEvent` are more about gaining access to internals of existing classes whereas I wanted the `ALGMockedObject` to really be just a dumb call-recorder. I guess we can rename them all to "Fake" for consistency, but then we should think of another name than "FakedObject" or add a class comment

Olliwehr replied:
Ahh okay, yeah good point. Then i think its better to go with "Mocked" here as it is now 👍

niklasmohrin replied:
I renamed the two algtypes but kept the MockedObject for now

Olliwehr commented on packages/AlgernonTests.package/ALGTestAction.class/instance/testWindowOperations.st:
```diff
@@ -0,0 +1,14 @@
+testing
+testWindowOperations
```
I suggest putting those two new test cases in the `ALGTestTypes` test class (category `testing window type` maybe?) as at least for me that is what we do here (rather than the the other tests in `ALGTestAction` that mostly test the creation of Algeractions). What's your opinion on that?

Olliwehr commented on packages/AlgernonTests.package/ALGTestAction.class/instance/testWindowOperations.st:
```diff
@@ -0,0 +1,14 @@
+		closeAllWindows;
+		closeWithoutChanges;
+		collapseAllWindows.
+	self assert: 3 equals: instance mockedDockingBar calls keys size.
```
The current coverage report states that your newly introduced indirections for the docking bar/the world in the WindowType and the WindowOperationsType aren't covered. 

I question if that is still the case if we use `dockingBar` instead of `mockedDockingBar` here. That won't make a difference in behaviour as you overwrote the dockingBar shorthand in the MockedWindowOperationsType with this indirection, right? 😄 

If this will lead to a coverage of the `dockingBar` shorthand then the same could be applied to the `world` shorthand here and the `dockingBar` one in the WindowType. Alltogether i think  this would bring us back over 99% again 🎉

niklasmohrin replied:
Well, I didn't think of that :laughing: 

I wouldn't change `mockedDockingBar` to just `dockingBar` here, because it is not what wanted to express. The test doesn't _really_ care about docking bars or in which way the window operations get them, it just wants to check up on the mock object. I think if we changed this to `dockingBar`, we would also run into the method of the mocked object and not into the one that is currently not covered.

I think we could add a test that verifies the interface of whatever the `dockingBar` and `world` methods return. Maybe get the object and then assert that it can understand the messages that we need it to understand (`closeAllWindows` etc.). This way, we would have the coverage and the test would be meaningful (maybe we change the docking bar at some point? don't know if that is possible :D)

What do you think?

Olliwehr replied:
I see your point. Didn't thought about whether that would be really meaningful. Was more like an approach based on pure coverage 😄 

Anyways, the 2 latest commits got us back over 99% regardless :D

Commit:
Remove some more unused methods

Commit:
Forward another message from the InputWrapper to the InputField

Before we introduced the InputWrapper the message was covered from the tests.
I hope this gets it covered again :)

Commit:
Moved two window type tests from ALGTestAction to ALGTestTypes

niklasmohrin committed:
rename mock -> fake

niklasmohrin committed:
add tests that ensure interface of  `world` and `dockingBar` indirection

bewee commented on packages/AlgernonTests.package/ALGTestTypes.class/instance/testWindowOperations.st:
```diff
@@ -0,0 +1,14 @@
+	self assert: 3 equals: instance mockedDockingBar calls keys size.
+	self assert: 1 equals: instance mockedWorld calls keys size.
+	instance mockedDockingBar calls values do: [ :coll | self assert: 1 equals: coll size ].
+	instance mockedWorld calls values do: [ :coll | self assert: 1 equals: coll size ]
```
Maybe consider giving `coll` a more descriptive name?

bewee commented on packages/AlgernonTests.package/ALGMockedObject.class/instance/recordCall..st:
```diff
@@ -0,0 +1,7 @@
+	(self calls
+		at: aMessage selector
+		ifAbsentPut: [ OrderedCollection new ])
+			add: aMessage
```
I find this a little odd to read, maybe we could add a temporary?

bewee commented on packages/AlgernonTests.package/ALGTestTypes.class/instance/testWindowTypeDockingBarMock.st:
```diff
@@ -0,0 +1,10 @@
+		closeAllWindowsLikeThis;
+		closeAllWindowsButThis.
+	self assert: 2 equals: instance mockedDockingBar calls keys size.
+	instance mockedDockingBar calls values do: [ :coll | self assert: 1 equals: coll size ]
```
Same here with `coll`

bewee reviewed (approved):
Great stuff!

niklasmohrin committed:
stuff

TheGrayStone committed:
Temporarily remove BOALGTest to check if it works then

TheGrayStone committed:
Add BOALGTest back in but without ensure internet connection

TheGrayStone committed:
I think I've got it

TheGrayStone committed:
Add ensure internet connection back in

TheGrayStone committed:
This maybe?

TheGrayStone committed:
Merge branch 'development' into coverage

niklasmohrin committed:
add totally sane way of communicating current git branch to the test runner

niklasmohrin committed:
separate baseline tests from normal tests, overwrite method for current git branch in CI script

niklasmohrin committed:
yaml

niklasmohrin committed:
yaml2

niklasmohrin committed:
stupid mac sed

niklasmohrin committed:
nicer names for jobs

TheGrayStone reviewed (approved):
Yes, finally!

niklasmohrin reviewed (commented):
I really like what you have done here

niklasmohrin committed:
I really trusted your review, nils
